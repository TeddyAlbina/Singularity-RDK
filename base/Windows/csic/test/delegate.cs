using System;
delegate void D(int x);
class A {
	public static void A1(int i) { Console.WriteLine("A1: {0}", i); }
	public void A2(int i) { Console.WriteLine("A2: {0}", i); }
}
class T {
	public static void Main() {
		D x = new D(A.A1);
		x(1);
		A t = new A();
		D y = new D(t.A2);
		y(2);
		D z = new D(y);
		z(3);
		D q = x + y; q(11);
		q += z; q(12);
		q -= x; q(13);
		D r = new D(q - y + x); r(14);
	}
}