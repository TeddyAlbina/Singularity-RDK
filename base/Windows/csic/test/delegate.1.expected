A1: 1
A2: 2
A2: 3
A1: 11
A2: 11
A1: 12
A2: 12
A2: 12
A2: 13
A2: 13
A2: 14
A1: 14
